rhoas-client/src/main/java/com/redhat/service/bridge/rhoas/AbstractAppServicesClientImpl.java

```diff
@@ -0,0 +1,100 @@
+    }
+
+    protected Uni<Void> topicsVoidCall(ApiVoidExecutor<TopicsApi> executor) {
+        return executeBlocking(() -> executeAndHandleException(new TopicsApi(defaultInstanceClient()), executor)).replaceWithVoid();
+    }
+
+    protected <T> Uni<T> securityValueCall(ApiValueExecutor<SecurityApi, T> executor) {
+        return executeBlocking(() -> executeAndHandleException(new SecurityApi(defaultMgmtClient()), executor));
+    }
+
+    protected Uni<Void> securityVoidCall(ApiVoidExecutor<SecurityApi> executor) {
+        return executeBlocking(() -> executeAndHandleException(new SecurityApi(defaultMgmtClient()), executor)).replaceWithVoid();
+    }
+
+    private com.openshift.cloud.api.kas.invoker.ApiClient defaultMgmtClient() {
+        com.openshift.cloud.api.kas.invoker.ApiClient defaultClient = com.openshift.cloud.api.kas.invoker.Configuration.getDefaultApiClient();
+        defaultClient.setBasePath(getBasePath());
+
+        // Configure HTTP bearer authorization: Bearer
+        com.openshift.cloud.api.kas.invoker.auth.HttpBearerAuth bearer = (com.openshift.cloud.api.kas.invoker.auth.HttpBearerAuth) defaultClient.getAuthentication("Bearer");
+        bearer.setBearerToken(getAccessToken());
+
+        return defaultClient;
+    }
+
+    private com.openshift.cloud.api.kas.auth.invoker.ApiClient defaultInstanceClient() {
+        com.openshift.cloud.api.kas.auth.invoker.ApiClient defaultClient = com.openshift.cloud.api.kas.auth.invoker.Configuration.getDefaultApiClient();
+        defaultClient.setBasePath(getBasePath());
+
+        // Configure OAuth2 access token for authorization: Bearer
+        com.openshift.cloud.api.kas.auth.invoker.auth.OAuth bearer = (com.openshift.cloud.api.kas.auth.invoker.auth.OAuth) defaultClient.getAuthentication("Bearer");
+        bearer.setAccessToken(getAccessToken());
+
+        return defaultClient;
+    }
+
+    private <T> Uni<T> executeBlocking(Supplier<T> supplier) {
+        return vertx.executeBlocking(Uni.createFrom().item(supplier));
+    }
+
+    private <A, T> T executeAndHandleException(A apiInstance, ApiValueExecutor<A, T> executor) {
+        try {
+            return executor.execute(apiInstance);
+        } catch (com.openshift.cloud.api.kas.invoker.ApiException e) {
+            throw new AppServicesException("Error when calling Kafka mgmt " + apiInstance.getClass().getSimpleName(), e);
+        } catch (com.openshift.cloud.api.kas.auth.invoker.ApiException e) {
+            throw new AppServicesException("Error when calling Kafka instance " + apiInstance.getClass().getSimpleName(), e);
+        }
+    }
+
+    private <A> Object executeAndHandleException(A apiInstance, ApiVoidExecutor<A> executor) {
+        return executeAndHandleException(apiInstance, api -> {
+            executor.execute(apiInstance);
+            return null;
+        });
+    }
+
+    protected interface ApiValueExecutor<A, T> {
```

Comment:
Can you provide a comment as to what these interfaces do?

Reply:
Done :+1: